distributed/src/test/scala/asobu/distributed/End2EndSpec.scala

```diff
@@ -0,0 +1,95 @@
+package asobu.distributed
+
+import akka.actor.{ActorSystem, ActorRef}
+import akka.util.Timeout
+import asobu.distributed.gateway.{GatewayRouter, DefaultHandlerBridgeProps, Gateway}
+import asobu.distributed.service.{DistributedController, EndpointsRegistryClientImp, EndpointsRegistryClient}
+import asobu.distributed.util.SpecWithActorCluster
+import junit.framework.TestResult
+import play.api.libs.json.{JsNumber, JsString, Json}
+import play.api.mvc.Results._
+import concurrent.duration._
+import scala.util.{Random, Try}
+import play.api.http.HttpVerbs._
+import play.api.test.{PlaySpecification, FakeRequest}
+
+class End2EndSpec extends PlaySpecification with SpecWithActorCluster {
+  import End2EndSpec._
+  val gatewayAction = DistributedSystem.GatewayApp().action
+  Thread.sleep(1000)
+  (new DistributedSystem.ServiceBackend.App).initControllers
+  Thread.sleep(1000)
+
+  "GET /cats/:catId returns OK Json response" >> {
+    val req = FakeRequest(GET, "/cats/3")
+    val resp = gatewayAction(req)
+    status(resp) === OK
+    (contentAsJson(resp) \ "id").get === JsString("3")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/iHeartRadio/asobu/file/2373689183/issues/source?bid=3037259&fileBranchId=3224055#l27)